presto-native-execution/presto_cpp/main/connectors/arrow_flight/tests/ArrowFlightConnectorDataTypeTest.cpp

```diff
@@ -0,0 +1,328 @@
+             .flightTableScan(velox::ROW({"bool_col"}, {velox::BOOLEAN()}))
+             .planNode();
+
+  AssertQueryBuilder(plan)
+      .splits(makeSplits({"sample-data"}))
+      .assertResults(makeRowVector({boolVec}));
+}
+
+TEST_F(ArrowFlightConnectorDataTypeTest, integerTypes) {
+  std::vector<int8_t> tinyData = {
+      -128, 0, 127, std::numeric_limits<int8_t>::max()};
+  std::vector<int16_t> smallData = {
+      -32768, 0, 32767, std::numeric_limits<int16_t>::max()};
+  std::vector<int32_t> intData = {
+      -2147483648, 0, 2147483647, std::numeric_limits<int32_t>::max()};
+  std::vector<int64_t> bigData = {
+      -3435678987654321234LL,
+      0,
+      4527897896541234567LL,
+      std::numeric_limits<int64_t>::max()};
+
+  updateTable(
+      "sample-data",
+      makeArrowTable(
+          {"tinyint_col", "smallint_col", "integer_col", "bigint_col"},
+          {makeNumericArray<arrow::Int8Type>(tinyData),
+           makeNumericArray<arrow::Int16Type>(smallData),
+           makeNumericArray<arrow::Int32Type>(intData),
+           makeNumericArray<arrow::Int64Type>(bigData)}));
+
+  auto tinyintVec = makeFlatVector<int8_t>(tinyData);
+
+  auto smallintVec = makeFlatVector<int16_t>(smallData);
+
+  auto integerVec = makeFlatVector<int32_t>(intData);
+
+  auto bigintVec = makeFlatVector<int64_t>(bigData);
+
+  core::PlanNodePtr plan;
+  plan = ArrowFlightPlanBuilder()
+             .flightTableScan(velox::ROW(
+                 {"tinyint_col", "smallint_col", "integer_col", "bigint_col"},
+                 {velox::TINYINT(),
+                  velox::SMALLINT(),
+                  velox::INTEGER(),
+                  velox::BIGINT()}))
+             .planNode();
+
+  AssertQueryBuilder(plan)
+      .splits(makeSplits({"sample-data"}))
+      .assertResults(
+          makeRowVector({tinyintVec, smallintVec, integerVec, bigintVec}));
+}
+
+TEST_F(ArrowFlightConnectorDataTypeTest, realType) {
+  std::vector<float> realData = {
+      std::numeric_limits<float>::min(),
+      0.0f,
+      3.14f,
+      std::numeric_limits<float>::max()};
+  std::vector<double> doubleData = {
+      std::numeric_limits<double>::min(),
+      0.0,
+      3.14159,
+      std::numeric_limits<double>::max()};
+
+  updateTable(
+      "sample-data",
+      makeArrowTable(
+          {"real_col", "double_col"},
+          {makeNumericArray<arrow::FloatType>(realData),
+           makeNumericArray<arrow::DoubleType>(doubleData)}));
+
+  auto realVec = makeFlatVector<float>(realData);
+  auto doubleVec = makeFlatVector<double>(doubleData);
+
+  core::PlanNodePtr plan;
+  plan = ArrowFlightPlanBuilder()
+             .flightTableScan(velox::ROW(
+                 {"real_col", "double_col"}, {velox::REAL(), velox::DOUBLE()}))
+             .planNode();
+
+  AssertQueryBuilder(plan)
+      .splits(makeSplits({"sample-data"}))
+      .assertResults(makeRowVector({realVec, doubleVec}));
+}
+
+TEST_F(ArrowFlightConnectorDataTypeTest, varcharType) {
```

Comment:
Please add tests with length > 12 since that is out of line. Also add tests with special characters and varbinary.

Reply:
What is the issue with length > 12?

Reply:
ah, I see.. > 12 then a reference to the string is held